### DISCUSSION TITLE
Unofficial Homebrew tap for MacOS

### DISCUSSION THREAD

**Discussion by vancluever** [input-leap/input-leap#1623] [Show and tell]
Hey folks!

Building on the comments in #1517, I've made a Homebrew tap that you can use to install on MacOS in the interim, until there's an official formula or cask out.

https://github.com/vancluever/homebrew-input-leap

To use the tap, just run `brew tap vancluever/input-leap` and then `brew install input-leap`.

Binaries, man pages, and documentation should all be correctly put in their right places. Note as this is not a cask, you will have to run the `InputLeap.app` application directly. The path you want to use is included in the caveats for your convenience!

I'll keep an eye out for updates and update the formula regularly, especially when there's any particularly big updates. But if I miss one, feel free to put in a PR!

**Comment by Andrew-J-Larson** [comment]
That's great man, I'll try this out tomorrow.

**Comment by earthsound** [comment]
Running macOS 13.6.1 on an Intel iMac. 

I get the following error when running `brew install input-leap` after adding the tap:
```
-- Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR) 
CMake Error at CMakeLists.txt:90 (message):
  Missing library: curl

-- Configuring incomplete, errors occurred!
```

I ran the following:
`brew install curl`
`echo 'export PATH="/usr/local/opt/curl/bin:$PATH"' >> /Users/username/.bash_profile`
`export LDFLAGS="-L/usr/local/opt/curl/lib"`
`export CPPFLAGS="-I/usr/local/opt/curl/include"`
`export PKG_CONFIG_PATH="/usr/local/opt/curl/lib/pkgconfig"`
`export HOMEBREW_FORCE_BREWED_CURL=1`

...but I'm still getting this error. How can I point cmake to the curl libraries?

The first thing in my `$PATH` is `/usr/local/opt/curl/bin`.

**Reply by vancluever** [reply]
You may find some info in https://github.com/vancluever/homebrew-input-leap/issues/4 and https://github.com/input-leap/input-leap/discussions/1678. The gist of it is that you should not need to install curl or do anything different than what is already installed. If so something is up with the build environment where it cannot find curl through the dynamic linker cache.

**Reply by earthsound** [reply]
Thanks! Symlinking did the trick.

**Comment by the-spyke** [comment]
@vancluever Could you please update the formula because there are now Apple Silicon builds: https://github.com/input-leap/input-leap/discussions/1793#discussioncomment-8577620

**Reply by vancluever** [reply]
👀 I just checked the universal binary (the one we should probably bundle) and noticed that it's not signed or notarized. @sithlord48 I know this is pre-release, but are there any plans to do this? This should hopefully be resolved before release as well.

Regardless of this, this means that the tap can be switched to a Cask, which is great news anyway and something I'll for sure work on!

**Reply by sithlord48** [reply]
As signing requires you to pay apple for a cert I doubt this will be happening until we at least get a real release. Also not sure if we have a cert or who will pay for it.

**Reply by vancluever** [reply]
Just checked and I'm not too sure if being able to source the release off of the builds page is going to work here (I don't think the links are stable, they 404 when you follow them directly), so we might need to continue to use the VCS build for now until releases are actually published.

**Reply by shymega** [reply]
The links should be stable - it might be because you have to be signed-in to download them. Are you using the API to get a direct asset link?

**Reply by Andrew-J-Larson** [reply]
Yes, the artifact downloads require to be signed in (which is a pain imo).

**Reply by shymega** [reply]
Unfortunately, it's out of our control.

@joshin42 Regarding that fee-waiver, we're not eligible. I for one don't own a Mac, and don't want to go down the Hackintosh route either, so I'm reluctant to support any packaging for Macs that I can't troubleshoot.

**Reply by sithlord48** [reply]
This will be resolved when we do "releases" from those artifacts. those links will not require sign in and should not move after wards for now your best bet for homebrew is to build from source.

**Reply by vancluever** [reply]
I've refreshed the build for now to 6cfeacd. 

One thing that I should mention - I haven't been using input-leap myself personally lately and I haven't been doing too much development on MacOS at the moment. As such updates to the tap will probably be sparse unless requested. PRs are welcome, or if anyone is interested in just taking it over so that there's a better cadence, let me know!

**Comment by joshin42** [comment]
Apple does have a fee-waiver program.

https://developer.apple.com/support/membership-fee-waiver/

**Comment by dlozynski** [comment]
If anyone want to test, I produced cask as formula was not needed anymore. Never the less is based on vancluever tap and then just upgraded few times. If time will allow I will try to patch it to official cask repo
https://github.com/dlozynski/homebrew-tools

**Reply by vancluever** [reply]
@dlozynski thanks for posting on this issue as I haven't used input-leap in quite a while and I didn't know the progress that's been made! With this in mind I'm just going to archive my repository. Glad it was a good reference!

**Comment by sithlord48** [comment]
You could always try our [Deskflow homebrew tap ](https://github.com/deskflow/homebrew-tap)
